### PR TITLE
feat(components): Add condensed styles to Table

### PR DESCRIPTION
**Pull request by marielakas**
Addresses https://sumupteam.atlassian.net/browse/PMNTSS-14

## Purpose

This PR adds condensed styles to the Table component. Check out the design below: 

Screenshot 2019-08-01 at 15 14 56

## Approach and changes

I've added the `condensed` property which enables the new styles in the Table and tweaked the `TableHeader` and `TableCell` components to meet the design requirements. The next step would be to make the table body scrollable but I've decided to make this in a separate PR so reviewing is easier :))

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [ ] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/components/Table/components/TableBody/index.js [hunk 1]
```diff
@@ -91,6 +93,11 @@ TableBody.propTypes = {
    * Handled internally
    */
   sortHover: PropTypes.number,
+  /**
+   * [PRIVATE] Adds condensed styles to the table.
```
Comment (connor-baer) [inline src/components/Table/components/TableBody/index.js]: I'd like to stick to the [JSDoc](https://devdocs.io/jsdoc/tags-private) conventions here :) 

```suggestion
   * @private Adds condensed styles to the table.
```

File: src/components/Table/components/TableHeader/index.js [hunk 2]
```diff
@@ -144,6 +162,11 @@ TableHeader.propTypes = {
    * Handled internally
    */
   fixed: PropTypes.bool,
+  /**
+   * [PRIVATE] Adds condensed style to the Header based on the table props.
```
Comment (connor-baer) [inline src/components/Table/components/TableHeader/index.js]: Same as above :)
```suggestion
   * @private Adds condensed style to the Header based on the table props.
```

File: src/components/Table/components/TableHead/index.js [hunk 3]
```diff
@@ -79,6 +90,11 @@ TableHead.propTypes = {
    * Enables/disables sticky columns on mobile
    */
   rowHeaders: PropTypes.bool,
+  /**
+   * [PRIVATE] Adds condensed styles the table according to the table props.
+   * Handled internally.
+   */
+  condensed: PropTypes.bool,
   /**
    * [PRIVATE] sortBy handler
```
Comment (connor-baer) [inline src/components/Table/components/TableHead/index.js]: Hm, I see that this format has already been used elsewhere 🤔 Which one do you prefer? Would you mind updating these to the `@private` syntax as well?

**Comment by marielakas** [comment]
@fernandofleury Yup, it's looking the same as before. I haven't implemented the condensed version for mobile because I have to check with design if it's okay to make the cells smaller there.

File: src/components/Table/components/TableHead/index.js (hunk 3 above)
Comment (marielakas) [reply src/components/Table/components/TableHead/index.js]: Done! :)

**Pull request merged**